### PR TITLE
DAT 3732 Infobox builder header section editing

### PR DESCRIPTION
**Pull request by RafLeszczynski**
## Links

* https://wikia-inc.atlassian.net/browse/DAT-3732
* https://github.com/Wikia/app/pull/9832

## Description

Add ability to edit section header element using infobox builder UI

## Reviewers
@dianafa

### CODE REVIEW

File: front/main/app/components/infobox-builder-edit-item-section-header.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		set(key, value) {
+			this.get('editSectionHeaderItem')(this.get('item'), {collapsible: value});
+			return value;
+		}
+	}),
+
+	init() {
+		this._super(...arguments);
+		this.isHelpVisible = false;
+		this.classNames = ['sidebar-content-padding'];
```
Comment (vforge) [inline front/main/app/components/infobox-builder-edit-item-section-header.js]: You don't need to assign values in `init()` function in this case - see https://github.com/Wikia/mercury/pull/2055/files#diff-eefbfc8c2642ba8b1f8be514c7cdb243R11 below.

Comment (hakubo) [inline front/main/app/components/infobox-builder-edit-item-section-header.js]: https://poteto.github.io/component-best-practices/#/leaky

File: front/main/app/components/infobox-builder-edit-item-row.js [hunk 2]
```diff
@@ -1,25 +1,17 @@
...
+		labelValue: Ember.computed('item.data.label', {
+			get() {
+				return this.get('item.data.label');
+			},
+			set(key, value) {
+				this.get('editRowItem')(this.get('item'), value);
+				return value;
+			}
+		})
 	}
```
Comment (dianafa) [inline front/main/app/components/infobox-builder-edit-item-row.js]: don't we need anymore `showHelp` here?

File: front/main/app/components/infobox-builder-edit-item-row.js [hunk 3]
```diff
@@ -1,25 +1,17 @@
...
-
-	actions: {
-		showHelp() {
-			this.set('isHelpVisible', true);
-		}
+export default Ember.Component.extend(
+	InfoboxBuilderEditItemMixin,
+	{
+		labelValue: Ember.computed('item.data.label', {
+			get() {
```
Comment (dianafa) [inline front/main/app/components/infobox-builder-edit-item-row.js]: shouldn't we add here also `editRowItem` and `item` ?

File: front/main/tests/unit/models/infobox-builder-test.js [hunk 4]
```diff
@@ -542,3 +602,76 @@ test('extend image data', (assert) => {
...
+				},
+				expected: {
+					data: '',
+					collapsible: true
+				}
+			},
+			{
+				newValues: null,
+				expected: {
+					data: defaultName,
```
Comment (dianafa) [inline front/main/tests/unit/models/infobox-builder-test.js]: if would be nice to add here case when data is `null`, data is `{}` and data is `''` and newValues is `{}`. what do you think?

File: front/main/app/components/infobox-builder-edit-item-row.js (hunk 3 above)
Comment (dianafa) [inline front/main/app/components/infobox-builder-edit-item-row.js]: we shouldn't :) :+1:

File: front/main/tests/unit/models/infobox-builder-test.js (hunk 4 above)
Comment (RafLeszczynski) [inline front/main/tests/unit/models/infobox-builder-test.js]: done

**Comment by dianafa** [comment]
LGTM

**Pull request merged**